### DISCUSSION TITLE
My plugin is not updating automatically with drawing turns on.

### DISCUSSION THREAD

**Discussion by nilsi** [FelixKratz/SketchyBar#511] [Q&A] [answered]
I build a plugin that either shows 1, *, or nothing depending on the status of the notification from the Slack app.

It works automatically when it goes from showing a notification to hiding it (after reading in Slack.)

However when there is a new notification it does not update unless I restart SketchyBar or run the plugin with `./plugins/slack.sh`
 
Here the code:

./plugins/slack.sh

```
#!/bin/bash

SLACK_INFO=$(lsappinfo info -only StatusLabel `lsappinfo find LSDisplayName=Slack`)
COUNT=${SLACK_INFO:25:1}

if [ $COUNT = "\"" ]; then
  DRAWING=off
else
  DRAWING=on
fi

sketchybar --set slack drawing=$DRAWING label="${COUNT}"
```

items/slack.sh

```
#!/bin/bash

slack=(
    update_freq=3
    icon.drawing=off
    label.font="$FONT:Semibold:14.0"
    script="$PLUGIN_DIR/slack.sh"
)

sketchybar --add item slack right        \
           --set slack "${slack[@]}"  \
```

I'm not very experienced with bash or SketchyBar so any help would be awesome. Thank you!

**Comment by FelixKratz** [comment, marked as answer]
Add `updates=on` to the slack properties. You have probably set it as a default to `when_shown`, thus scripts only get executed when the item is drawn to screen.

**Reply by nilsi** [reply]
That worked! Thanks a lot :)
